runnc-cont/runnc_cont.go

```diff
@@ -31,21 +30,75 @@ import (
 	"github.com/nabla-containers/runnc/nabla-lib/storage"
 )
 
-type arrayEnvVars []string
+type RunncCont struct {
+	// NablaRunBin is the path to 'nabla-run' binary.
+	NablaRunBin string
 
-func (i *arrayEnvVars) String() string {
-	return "my string representation"
-}
+	NablaRunArgs []string
 
-func (i *arrayEnvVars) Set(value string) error {
-	*i = append(*i, value)
-	return nil
+	// UniKernelBin is the path to 'unikernel' binary.
+	UniKernelBin string
+
+	// Tap tap device. (e.g. tap100)
+	Tap string
+
+	IPAddress net.IP
+	IPMask    net.IPMask
+	GateWay   net.IP
+
+	// Memory max memory size in MBs.
+	Memory int64
+
+	// IsInDocker means running in a Docker container or not.
+	IsInDocker bool
+
+	// IsInKubernetes means running in a Kubernetes cluster or not.
+	IsInKubernetes bool
+
+	// Disk is the path to disk
+	Disk string
+
+	// WorkingDir current working directory.
+	WorkingDir string
+
+	// Env is a list of environment variables.
+	Env []string
 }
 
-var envVars arrayEnvVars
+// NewRunncCont returns a brand new runnc-cont
+func NewRunncCont(cfg Config) (*RunncCont, error) {
+	if len(cfg.IPAddress) == 0 {
+		cfg.IPAddress = "10.0.0.2"
```

Comment:
`IsInKubernetes` takes precedence over this, so it's ok :). This will be good for when we can add a flag to `runnc` to tell it that it is running standalone so that it can do it's own network and not have that weird hack in directly calling `runnc` in the tests right now (with pre-start hooks). But that's a scope of another PR, so this is great for now!